event_routing_backends/processors/caliper/event_transformers/problem_interaction_events.py

```diff
@@ -87,27 +102,35 @@ def get_object(self):
                 self.get_data('data'),
                 self.get_data('context.course_id')
             )
+        key = self.get_event_name_key()
+
+        if key == 'showanswer':
+            object_id = object_id+'/solution'
+        elif key == 'edx.problem.hint.demandhint_displayed':
+            object_id = object_id+'/hint/'+str(event_data.get('hint_index', ''))
+        elif key == 'problem_check_server':
+            object_id = object_id+'/user/'+get_anonymous_user_id(self.extract_username_or_userid())\
+                        + '/attempt/'+str(event_data.get('attempts', ''))
 
         caliper_object = self.transformed_event['object']
         caliper_object.update({
-            'id': object_id,
-            'type': OBJECT_TYPE_MAP[self.get_data('name', True)],
+            'id': self.get_object_iri('xblock', object_id),
+            'type': OBJECT_TYPE_MAP[key],
+            'name': OBJECT_NAME_MAP[key],
         })
 
-        if event_data and isinstance(event_data, dict):
-            extensions = self.extract_subdict_by_keys(
-                event_data, [
-                    'module_id',
-                    'grade',
-                    'max_grade',
-                    'success',
-                    'attempts',
-                    'event_transaction_id',
-                    'event_transaction_type',
-                    'weighted_earned',
-                    'weighted_possible',
-                ]
+        if key == 'problem_check_server':
```

Comment:
This `if` can be removed and code can be moved inside `elif key == 'problem_check_server':` block above. also we can define a variable `anonymous_user_id` and use it at multiple places to avoid multiple db calls.
`anonymous_user_id = get_anonymous_user_id(self.extract_username_or_userid())`